frontend/src/components/form/BodyModification.tsx

```diff
@@ -1,76 +1,76 @@
   formatLabel: (text: string) => string;
 }
 
+type BodyModificationFieldArray = {
+  [name: string]: Array<{
+    location: string;
+    description?: string | null;
+  }>;
+};
+
 const CLASSNAME = "BodyModification";
 
 const BodyModification: React.FC<BodyModificationProps> = ({
   name,
   locationPlaceholder,
   descriptionPlaceholder,
   control,
-  defaultValues,
   formatLabel,
 }) => {
-  const [modifications, setModifications] = useState(defaultValues || []);
-  const selectRef = useRef(null);
-
+  const {
+    fields: modifications,
+    append,
+    remove,
+    update,
+  } = useFieldArray<BodyModificationFieldArray, string, "location">({
+    control,
+    name,
+    keyName: "location",
+  });
   const isNewLocationValid = (inputValue: string): boolean =>
     !!inputValue &&
     !modifications.find(({ location }) => inputValue === location);
 
   const handleNewLocation = (inputValue: string) => {
-    setModifications([...modifications, { location: inputValue }]);
+    append({ location: inputValue });
   };
 
-  const removeMod = (index: number) =>
-    setModifications(modifications.filter((_, i) => i !== index));
-
-  const modificationList = modifications.map((mod, index) => {
-    const idx = `${name}[${index}]`;
-    return (
-      <Form.Row key={mod.location} className="mb-1">
-        <InputGroup className="col">
-          <InputGroup.Prepend>
-            <InputGroup.Text className="font-weight-bold">
-              Location
-            </InputGroup.Text>
-          </InputGroup.Prepend>
-          <Controller
-            render={({ field }) => <Form.Control {...field} readOnly />}
-            name={`${idx}.location`}
-            control={control}
-            defaultValue={mod.location}
-          />
-          <Controller
-            render={({ field }) => (
-              <Form.Control {...field} placeholder={descriptionPlaceholder} />
-            )}
-            name={`${idx}.description`}
-            defaultValue={mod.description}
-            control={control}
-          />
-          <InputGroup.Append>
-            <Button variant="danger" onClick={() => removeMod(index)}>
-              Remove
-            </Button>
-          </InputGroup.Append>
-        </InputGroup>
-      </Form.Row>
-    );
-  });
+  const modificationList = modifications.map((mod, index) => (
+    <Form.Row key={mod.location} className="mb-1">
+      <InputGroup className="col">
+        <InputGroup.Prepend>
+          <InputGroup.Text className="font-weight-bold">
+            Location
+          </InputGroup.Text>
+        </InputGroup.Prepend>
+        <Form.Control defaultValue={mod.location} readOnly />
+        <Form.Control
+          defaultValue={mod.description ?? ""}
+          placeholder={descriptionPlaceholder}
+          onInput={(e: React.ChangeEvent<HTMLInputElement>) =>
+            update(index, { ...mod, description: e.currentTarget.value })
+          }
```

Comment:
There's probably a better way to do that.

Reply:
You should be able to use a `<Controller>` as well, but this is very straightforward so no complaints from me.